util/httputil/http_util.go

```diff
@@ -44,6 +46,36 @@ func HttpGet(url string, auth *config.Auth, timeout time.Duration) ([]byte, int,
 	return body, resp.StatusCode, err
 }
 
+func HttpPost(ctx context.Context, url string, auth *config.Auth, body io.Reader) ([]byte, int, error) {
```

Comment:
This is slightly different from `HttpGet` in that it accepts a `context` as its first arg. I thought it'd be nice to passthrough the request.Context to propagate cancellation but I'm happy to update this to look more like `HttpGet`.

Reply:
I'd propose as you say, to keep it similar as "HttpGet" first
![image](https://user-images.githubusercontent.com/1662329/132383838-07ed614f-0434-4676-9ea3-6c6ed39555ff.png)

To not introduce a refactoring in the use of these two close methods, as there would be several places.
Also, how to pass the context may vary, and probably a nice refactoring would touch more files, I think.

Reply:
Sure I'll update to be more like httpget and save the refactoring for another change. Updating all instances of httpget as well is definitely a larger change.